types/hasura/index.d.ts

```diff
@@ -0,0 +1,266 @@
+// Type definitions for Hasura GraphQL Engine API types 0.1
+// Project: https://github.com/hasura/graphql-engine
+// Definitions by: Toan Nguyen <https://github.com/hgiasac>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export const XHasuraAdminSecret = 'x-hasura-admin-secret';
+export const XHasuraClientName = 'hasura-client-name';
+export const XHasuraRole = 'x-hasura-role';
+export const XHasuraUserID = 'x-hasura-user-id';
+export const HASURA_ROLE_ADMIN = 'admin';
+
+export interface ColumnBoolExp<T = string | number | boolean> {
+    _eq?: T;
+    _neq?: T;
+    _gt?: T;
+    _gte?: T;
+    _lt?: T;
+    _lte?: T;
+    _is_null?: boolean;
+    _in?: T[];
+    _nin?: T[];
+}
+
+export interface TextColumnBoolExp extends ColumnBoolExp<string> {
+    _like?: string;
+    _ilike?: string;
+    _nlike?: string;
+    _nilike?: string;
+    _similar?: string;
+    _nsimilar?: string;
+}
+export type JSONColumnBoolExp = ColumnBoolExp<string>;
+export interface JSONBColumnBoolExp extends ColumnBoolExp<string> {
+    _contains?: string;
+    _contained_in?: string;
+    _has_key?: string;
+    _has_keys_all?: string;
+    _has_keys_any?: string;
+}
+
+type Equals<X, Y> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y ? 1 : 2 ? true : false;
```

Comment:
I think this is usually defined as
```ts
type Equals<X, Y> = [X] extends [Y] ? [Y] extends [X] ? true : false : false;
```

Then you don't need to suppress the lint rule.

Reply:
I tried your suggestion, however, below equality check don't work anymore
```ts
export type ScalarJSON<T> = T & {
    __type?: 'json';
}

hasura.Equals<{ a: string }, hasura.ScalarJSON<{ a: string }>> // true
```